### PR TITLE
TEN-122

### PR DESCRIPTION
**Pull request by jparekh59**

### CODE REVIEW

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/ContactSAController.scala [hunk 1]
```diff
@@ -23,23 +23,19 @@ import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
...
 
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class ReportSuspiciousIdController @Inject()(
-                                              authAction: AuthAction,
-                                              mcc: MessagesControllerComponents,
-                                              logger: EventLoggerService,
-                                              reportSuspiciousId: ReportSuspiciousID
+class ContactSAController @Inject()(
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/ContactSAController.scala:34]: I think we can have a clearer name for this page and controller. It's a reporting page that is relevant for if the user sees an account with a PT or SA Enrolment they don't recognise.  Something along the lines of `ReportAccount, ReportUserID` etc.

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/ContactSA.scala.html [hunk 2]
```diff
@@ -39,57 +39,59 @@
...
-
 @if(currentAccountHasSA) {
-    <p class="govuk-body">@messages("reportSuspiciousId.sa.text1") </p>
-    <br>
-    @govukButton(Button(
-    content = Text(messages("common.continue")),
-    href = Some("?")
-    ))
+<p class="govuk-body">@messages("contactSA.sa.text1") </p>
+<br>
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/ContactSA.scala.html:89]: I don't see this SA text anywhere in the current design.

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/ContactSA.scala.html [hunk 3]
```diff
@@ -39,57 +39,59 @@
...
+    @messages("contactSA.text1")
+    @messages("contactSA.text2")
+</p>
 
-<h1 class="govuk-heading-xl">@{messages(s"reportSuspiciousId.heading")}</h1>
 <dl class="govuk-summary-list">
 
-    @createRowWith(messages("reportSuspiciousId.userId"), ptAccountDetails.userId)
-    @createRowWith(messages("reportSuspiciousId.email"), ptAccountDetails.email.get)
-    @createRowWith(messages("reportSuspiciousId.lastSignedIn"), ptAccountDetails.lastLoginDate)
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/ContactSA.scala.html:58]: I assume this was an unintentional deletion? We still need the last signed in info.

File: test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/views/ContactSASpec.scala [hunk 4]
```diff
@@ -92,58 +92,43 @@ class ReportSuspiciousIDSpec extends TestFixture {
...
-      "includes the last signed in field with correct value" in {
-        suspiciousIdDetailsRows
-          .get(2)
-          .getElementsByClass(Selectors.summaryListKey)
-          .text() shouldBe "Last signed in"
-        suspiciousIdDetailsRows
-          .get(2)
-          .getElementsByClass(Selectors.summaryListValue)
-          .text() shouldBe accountDetails.lastLoginDate
-      }
```
Comment (Ufuoma-Daniel) [inline test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/views/ContactSASpec.scala:104]: Same here, this test shouldn't be deleted!

File: test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/views/ContactSASpec.scala [hunk 5]
```diff
@@ -153,42 +138,51 @@ class ReportSuspiciousIDSpec extends TestFixture {
...
         detailsBlockParagraphs
           .get(2)
-          .text() shouldBe ReportSuspiciousIDMessages.informationBlock(3)
+          .text() shouldBe ContactSAMessages.informationBlock(3)
+      }
+      "correct gov-uk link target and link text for Relay UK link" in {
+        doc(result)
+          .getElementsByClass(Selectors.links).get(1)
+          .attr("target") shouldBe "_blank"
+
```
Comment (Ufuoma-Daniel) [inline test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/views/ContactSASpec.scala:158]: Can we also verify the link href too?

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/ContactSA.scala.html (hunk 3 above)
Comment (jparekh59) [reply app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/ContactSA.scala.html:58]: It's intentional. as this is cover by logic created by Christine and in this view has been used. 
 @for(mfaDetail <- ptAccountDetails.mfaDetails) {
    @createRowWith(mfaDetail.factorName, mfaDetail.factorValue)

File: test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/views/ContactSASpec.scala (hunk 4 above)
Comment (jparekh59) [reply test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/views/ContactSASpec.scala:104]: depending on what is returned by mfa, at least one field of the mfa details ,the text message, is getting validated.
To ease the concern we are having a story for validating mfa details call and validating the returned value across the journey, so guess with in we can validate the permutations.

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/ContactSA.scala.html (hunk 2 above)
Comment (Ufuoma-Daniel) [reply app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/ContactSA.scala.html]: Below?

Comment (Ufuoma-Daniel) [reply app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/ContactSA.scala.html]: Apologies, I just realised where I got confused, I was looking at the message key for "contactSA.text1" and not  "contactSA.sa.text1".

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/ContactSA.scala.html [hunk 6]
```diff
@@ -65,6 +65,8 @@ <h1 class="govuk-heading-xl">@{messages(s"contactSA.heading")}</h1>
     @for(mfaDetail <- ptAccountDetails.mfaDetails) {
     @createRowWith(mfaDetail.factorName, mfaDetail.factorValue)
     }
+    @createRowWith(messages("contactSA.lastLoginDate"), ptAccountDetails.lastLoginDate)
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/ContactSA.scala.html]: Hate to be nitpicky but won't this place the last sign in date after the MFA info, the design wants it before right? Just after the email.

**Review by Ufuoma-Daniel — APPROVED** [review]

**Comment by Ufuoma-Daniel** [comment]
Unit tests for the order of the mfa details and last login stuff need updating and then it's all good.

**Review by Ufuoma-Daniel — APPROVED** [review]

**Pull request merged**